### PR TITLE
Grayson/airgap push progess

### PR DESCRIPTION
**Pull request by GraysonNull**

### CODE REVIEW

File: web/src/components/AirgapUploadProgress.jsx [hunk 1]
```diff
@@ -45,44 +46,53 @@ function AirgapUploadProgress(props) {
...
       </div>
     );
   }
 
-  props.data?.startPolling(2000);
-
+  props.data?.startPolling(1000);
+  
   let statusMsg = getAirgapInstallStatus?.currentMessage;
+  let jsonMessage;
```
Comment (salahalsaleh1993) [inline web/src/components/AirgapUploadProgress.jsx]: this can be moved inside the try block

File: web/src/components/AirgapUploadProgress.jsx [hunk 2]
```diff
@@ -45,44 +46,53 @@ function AirgapUploadProgress(props) {
...
+    jsonMessage = JSON.parse(statusMsg);
     const type = get(jsonMessage, "type");
     if (type === "progressReport") {
-      statusMsg = jsonMessage.compatibilityMessage;
-      // TODO: handle image upload progress here
+      try {
+        const parsedMsg = JSON.parse(jsonMessage.compatibilityMessage);
+        statusMsg = parsedMsg.compatibilityMessage;
+        processingImages = parsedMsg.images.sort((a, b) => (a.status > b.status) ? -1 : 1);
+        console.log(processingImages)
```
Comment (salahalsaleh1993) [inline web/src/components/AirgapUploadProgress.jsx]: clean console.log

File: api/src/controllers/kots/KotsAPI.ts [hunk 3]
```diff
@@ -441,7 +441,11 @@ export class KotsAPI {
           await statusServer.termination((resolve, reject, obj): boolean => {
             // Return true if completed
             if (obj.status === "running") {
-              request.app.locals.stores.kotsAppStore.setAirgapInstallStatus(obj.display_message, "running");
+              let objMsg = obj.display_message;
+              if (typeof objMsg === "string") {
```
Comment (salahalsaleh1993) [inline api/src/controllers/kots/KotsAPI.ts]: why not check if `objMsg` is parsable, and then create a new object if not? you won't have to update the frontend side to handle it then

Comment (salahalsaleh1993) [reply api/src/controllers/kots/KotsAPI.ts]: with something like
```
function isJson(value) {
    try {
      JSON.parse(value);
    } catch(err) {
      return false;
    }
  }
```

File: api/src/controllers/kots/KotsAPI.ts [hunk 4]
```diff
@@ -429,7 +429,7 @@ export class KotsAPI {
 
         const tmpDstDir = tmp.dirSync();
         try {
-          await request.app.locals.stores.kotsAppStore.setAirgapInstallStatus("Processing app package...", "running");
+          await request.app.locals.stores.kotsAppStore.setAirgapInstallStatus(JSON.stringify({ type: "progressReport", compatibilityMessage: "Processing app package..."}), "running");
```
Comment (divolgin) [inline api/src/controllers/kots/KotsAPI.ts]: This should be done in kots

File: api/src/controllers/kots/KotsAPI.ts [hunk 5]
```diff
@@ -454,7 +458,7 @@ export class KotsAPI {
             return false;
           });
 
-          await request.app.locals.stores.kotsAppStore.setAirgapInstallStatus("Creating app...", "running");
+          await request.app.locals.stores.kotsAppStore.setAirgapInstallStatus(JSON.stringify({ type: "progressReport", compatibilityMessage: "Creating app..."}), "running");
```
Comment (divolgin) [inline api/src/controllers/kots/KotsAPI.ts]: This should be done in kots too

File: web/src/components/AirgapUploadProgress.jsx [hunk 6]
```diff
@@ -45,44 +46,52 @@ function AirgapUploadProgress(props) {
...
     // Some of these messages will be JSON formatted progress reports.
-    const jsonMessage = JSON.parse(statusMsg);
+    let jsonMessage;
+    jsonMessage = JSON.parse(statusMsg);
     const type = get(jsonMessage, "type");
     if (type === "progressReport") {
-      statusMsg = jsonMessage.compatibilityMessage;
-      // TODO: handle image upload progress here
+      try {
+        const parsedMsg = JSON.parse(jsonMessage.compatibilityMessage);
```
Comment (salahalsaleh1993) [inline web/src/components/AirgapUploadProgress.jsx]: i think this is not needed anymore now that the api is updated. you can do `statusMsg = jsonMessage.compatibilityMessage;` and ignore on catch

File: web/src/components/AirgapUploadProgress.jsx [hunk 7]
```diff
@@ -45,44 +46,52 @@ function AirgapUploadProgress(props) {
...
-  props.data?.startPolling(2000);
-
+  props.data?.startPolling(1000);
+  
   let statusMsg = getAirgapInstallStatus?.currentMessage;
   try {
     // Some of these messages will be JSON formatted progress reports.
-    const jsonMessage = JSON.parse(statusMsg);
+    let jsonMessage;
+    jsonMessage = JSON.parse(statusMsg);
```
Comment (salahalsaleh1993) [inline web/src/components/AirgapUploadProgress.jsx]: this can be in 1 line

**Comment by GraysonNull** [comment]
[ch23126]

File: web/src/components/AirgapUploadProgress.jsx [hunk 8]
```diff
@@ -45,44 +46,47 @@ function AirgapUploadProgress(props) {
...
 
-  props.data?.startPolling(2000);
-
+  props.data?.startPolling(1000);
+  
   let statusMsg = getAirgapInstallStatus?.currentMessage;
   try {
     // Some of these messages will be JSON formatted progress reports.
     const jsonMessage = JSON.parse(statusMsg);
+    console.log(jsonMessage);
```
Comment (salahalsaleh1993) [inline web/src/components/AirgapUploadProgress.jsx]: clean console.log

Comment (GraysonNull) [reply web/src/components/AirgapUploadProgress.jsx]: :/ I removed this and apparently didn't save it before commiting...

Comment (salahalsaleh1993) [reply web/src/components/AirgapUploadProgress.jsx]: :)

**Pull request merged**